core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -233,6 +233,14 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
+   * Change the value of numExecutorsTarget.
+   */
+  def reSetNumExecutorsTarget(): Unit = {
+   logDebug(s"Now reset the value of numExecutorsTarget.")
+    numExecutorsTarget = conf.getInt("spark.dynamicAllocation.initialExecutors", minNumExecutors)
```

Comment:
I actually think we should reset on the driver side as this patch has done, since we end up calling `updateAndSync` anyway